### DISCUSSION TITLE
Redirect an url with params

### DISCUSSION THREAD

**Discussion by mediactive** [statamic/cms#7663] [General]
Hello!
How to keep the Google Click ID (GCLID) in the url following a redirect?
Example: https://www.bressesyndic.com/?gclid=xxxxxx redirected to https://www.bressesyndic.com/fr?gclid=xxxxxx
I thought something like 
```
Route::get('/{param?}', function ($param = 'gclid') {
    return redirect('/fr', ['param'=>$param])
});
```
 might work, but i get a error 500.

**Comment by jasonvarga** [comment]
What's the error?

**Reply by mediactive** [reply]
Here is the error message:

> Error 500 - Server Error
> The address you are trying to reach cannot be found or is unreachable at this time.
> 
> Hint:
> 
> Check the address you entered and correct it if necessary.
> Return to the Home Page
> Please report any technical issues.

If I use `Route::redirect('/', '/fr', 301);` instead of
```
Route::get('/{param?}', function ($param = 'gclid') {
    return redirect('/fr', ['param'=>$param])
});
```
i have no error but the new url doesn't get the gclid param.

**Reply by mediactive** [reply]
It seems to work this way:
```
Route::get('/', function () {
    $gclid = request()->input('gclid');
    return redirect('/fr?gclid='.$gclid);
});
```

**Reply by ryanmitchell** [reply]
You got it sorted but just so you know:

```
Route::get('/{param?}', function ($param = 'gclid') {
    return redirect('/fr', ['param'=>$param])
});
```

Would match urls like `yoururl.com/gclid`, not `yoururl.com/?gclid=xxx`
